src/porepy/grids/md_grid.py

```diff
@@ -297,11 +336,29 @@ def subdomain_to_boundary_grid(self, sd: pp.Grid) -> Optional[pp.BoundaryGrid]:
             sd: A subdomain in the mixed-dimensional grid.
 
         Returns:
-            The boundary grid associated with this subdomain. If ``sd`` is not on the
-            domain boundary, None is returned.
+            The boundary grid associated with this subdomain. ``None`` is returned if
+            ``sd`` is not in this mixed-dimensional grid, or the dimension of ``sd`` is
+            0.
 
         """
-        return self._subdomain_boundary_grids.get(sd, None)
+        return self._subdomain_to_boundary_grid.get(sd, None)
+
+    def boundary_grid_to_subdomain(self, bg: pp.BoundaryGrid) -> Optional[pp.Grid]:
```

Comment:
Depending on how we use the boundary grids in the model, we may or may not need this method. If it turns out we don't need it, let's try to remember to purge it again. We have fought hard to keep the model classes relatively clean.

It may be that it is obvious we will need this; if so, please bear with me.

Reply:
I have used this method in tests several times, for instance [here](https://github.com/pmgbergen/porepy/blob/e85d18487f19a72e71ab43de4a65904ff96d9b90/tests/unit/test_equation_system.py#L125). I think it'd be natural to make the property `_parent` of `BoundaryGrid` public through getter since we anyway store the related subdomain there. Then we would get rid of this method in the mdg.

Reply:
We agreed on making `_parent` public.